### PR TITLE
Add an option for CSV output

### LINKED ISSUE #32: Add a CSV output format
It'd be nice if the various test result functions (`PASS`, `FAIL`, etc) could emit CSV when requested.

For example, when requested, `FAIL` would go from this:

```
FAIL(example.com): No CAA records found
```

to this:

```csv
fail,example.com,No CAA records found
```

In terms of requesting CSV output, either a flag (`-c`?) or an environment variable (`TWA_FORMAT=csv`?) would be okay (or both?).

### PR DESCRIPTION
**Pull request by marado**
Closes #32.

### CODE REVIEW

File: twa [hunk 1]
```diff
@@ -78,34 +78,54 @@ function get_field {
       tolower(\$1) ~ /^${field}/ { print substr(\$0, index(\$0, \$2)) }" | tr -d '\r\n'
 }
 
+function output {
+  if [[ -n "${csv}" ]]; then
+    echo "${1},${2},${@:3}";
```
Comment (woodruffw) [inline twa]: Per `shellcheck`, this should probably be `${*:3}` here and below.

The explanation can also contain commas, so we probably need to double-quote it:

```bash
echo "${1},${2},\"${*:3}\""
```

**Comment by woodruffw** [comment]
Ah, you beat me to it on `@` vs `*` 😄. In that case, the only thing that needs to be done here is adding double quotes to the third field in the CSV output.

**Comment by marado** [comment]
> Ah, you beat me to it on `@` vs `*` smile. In that case, the only thing that needs to be done here is adding double quotes to the third field in the CSV output.

Isn't it already?
`echo "${1},${2},${*:3}";`

The double quotes are encompassing all three expandable arguments. Or do you mean something else?

**Comment by woodruffw** [comment]
> Or do you mean something else?

I meant in the output itself. Right now, if I do this:

```bash
FAIL "this is a message, with a comma"
```

It'll output:

```
FAIL,example.com,this is a message, with a comma
```

Which a CSV parser will interpret as 4 columns: `FAIL`, `example.com`, `this is a message`, and ` with a comma`. Instead, we want to emit something like this:

```
FAIL,example.com,"this is a message, with a comma"
```

Which a CSV parser will interpret correctly as 3 columns. We can do that by adding an additional set of escaped double quotes to the third field:

```bash
echo "${1},${2},\"${*:3}\""
```

**Comment by woodruffw** [comment]
Thanks! I'll merge this in a bit.

**Comment by woodruffw** [comment]
Thanks again, @marado!

**Pull request merged**